### PR TITLE
upgrade all the deps

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
-  - npm install -g bower
-  - npm install
-  - bower install
+  - yarn install --no-lockfile
 
 script:
-  - npm test
+  # Usually, it's ok to finish the test scenario without reverting
+  #  to the addon's original dependency state, skipping "cleanup".
+  - node_modules/.bin/ember try:one $EMBER_TRY_SCENARIO test --skip-cleanup
```
Comment (rwjblue) [inline .travis.yml]: I believe this will result in us not actually running the node tests?

File: .travis.yml [hunk 2]
```diff
@@ -1,22 +1,37 @@
...
-  - "7"
+
+sudo: false
+
+cache:
+  yarn: true
+
+env:
+  # we recommend testing LTS's and latest stable release (bonus points to beta/canary)
+  - EMBER_TRY_SCENARIO=ember-lts-2.4
```
Comment (rwjblue) [inline .travis.yml]: There is no ember specific code here, I don't think we need to test more than one (slowing down CI runs...)

File: .travis.yml [hunk 3]
```diff
@@ -1,22 +1,37 @@
 ---
-sudo: false
-
 language: node_js
 node_js:
-  - "4"
   - "6"
-  - "7"
```
Comment (rwjblue) [inline .travis.yml]: We need to continue to test against 4, 6, and 7.

Comment (stefanpenner) [inline .travis.yml]: Oops

**Comment by stefanpenner** [comment]
@rwjblue thanks for the review, should be addressed now!

**Pull request merged**